lib/core_dom/resource_url_resolver.dart

```diff
@@ -39,7 +39,15 @@ class ResourceUrlResolver {
   static final NodeTreeSanitizer _nullTreeSanitizer = new _NullTreeSanitizer();
   static final docForParsing = document.implementation.createHtmlDocument('');
 
-  static Node _parseHtmlString(String html) {
+  static String _getBaseUri() {
+    if (Uri.base.authority.isEmpty) {
+      throw "Uri.base.authority is Empty";
```

Comment:
Looks good.  The error message could be more descriptive.  e.g. "Relative URL resolution requires a valid base URI" or something so folks know that this is related to resolving relative URLs (won't be thrown otherwise.)